src/userDashboard/UserDashboard.js

```diff
@@ -32,21 +32,24 @@ export default function UserDashboard({ api }) {
   const [allWordsData, setAllWordsData] = useState(null);
   const [allWordsDataPerMonths, setAllWordsDataPerMonths] = useState({});
   const [referenceDate, setReferenceDate] = useState(new Date());
-  const [
-    dailyExerciseAndReadingTimes,
-    setDailyExerciseAndReadingTimes,
-  ] = useState(null);
-  const [
-    monthlyExerciseAndReadingTimes,
-    setMonthlyExerciseAndReadingTimes,
-  ] = useState({});
+  const [dailyExerciseAndReadingTimes, setDailyExerciseAndReadingTimes] =
+    useState(null);
+  const [monthlyExerciseAndReadingTimes, setMonthlyExerciseAndReadingTimes] =
+    useState({});
+
+  function handleChangeReferenceDate(newDate) {
+    setReferenceDate(newDate);
+    api.logUserActivity(api.USER_DASHBOARD_DATE_CHANGE, "", newDate);
+  }
 
   function handleActiveTabChange(tabId) {
     setActiveTab(tabId);
+    api.logUserActivity(api.USER_DASHBOARD_TAB_CHANGE, "", tabId);

```

Comment:
for now it's ok; but ten months from now, if we might change the order of the tabs, the tab number will be misleading. probably tab name would have been more future-proof.